### PR TITLE
Contact API updates (If no Tribol)

### PR DESCRIPTION
**Pull request by chapman39**
In the future we may want to test `TRIBOL_DIR=''` in Serac rather than Smith. Or always include tribol?

### CODE REVIEW

**Comment by jamiebramwell** [comment]
@ebchin , can you take a quick look at this?

**Comment by jamiebramwell** [comment]
> In the future we may want to test `TRIBOL_DIR=''` in Serac rather than Smith. Or always include tribol?

We should move the no-tribol tests here to catch issues like this.

**Comment by jamiebramwell** [comment]
Per @chapman39 , the changes introduced in https://github.com/LLNL/serac/pull/982/files were not propagated to the no-tribol sections.

**Review by jamiebramwell — APPROVED** [review]
Thanks @chapman39 !

File: src/serac/physics/contact/contact_data.cpp [hunk 1]
```diff
@@ -328,12 +328,14 @@ void ContactData::updateDofOffsets() const
   offsets_up_to_date_ = true;
 }
 
+}  // namespace serac
```
Comment (white238) [inline src/serac/physics/contact/contact_data.cpp]: Are we sure this is right? the `namespace serac` is inside the if of the ifdef then reopens in the else

Comment (white238) [reply src/serac/physics/contact/contact_data.cpp]: Pretty sure it wasnt right to begin with either... Can you move all namespacing outside of the ifdef?

Comment (chapman39) [reply src/serac/physics/contact/contact_data.cpp]: im pretty sure its right but it is weird to have namespace serac in both ifdef blocks.

Comment (chapman39) [reply src/serac/physics/contact/contact_data.cpp]: fixed

File: src/serac/physics/contact/contact_data.hpp [hunk 2]
```diff
@@ -156,13 +156,22 @@ class ContactData {
...
 
+#ifdef SERAC_USE_TRIBOL
   /**
    * @brief Have there been contact interactions added?
    *
    * @return true if contact interactions have been added
    * @return false if there are no contact interactions
    */
   bool haveContactInteractions() const { return !interactions_.empty(); }
+#else
```
Comment (white238) [inline src/serac/physics/contact/contact_data.hpp:169]: would it be better to put the ifdef inside of the function?

**Review by white238 — APPROVED** [review]

File: src/serac/physics/contact/contact_data.hpp (hunk 2 above)
Comment (chapman39) [reply src/serac/physics/contact/contact_data.hpp:169]: Yeah, I was doing this quickly. I'll update that

**Comment by chapman39** [comment]
Fixes #1055

**Comment by ebchin** [comment]
Thanks for fixing this @chapman39 !

**Pull request merged**